packages/devtools-reps/src/object-inspector/component.js

```diff
@@ -126,16 +129,12 @@ class ObjectInspector extends Component<Props> {
     // - OR the expanded paths number did not changed, but old and new sets
     //      differ
     // - OR the focused node changed.
-    return (
-      loadedProperties.size !== nextProps.loadedProperties.size ||
-      (expandedPaths.size !== nextProps.expandedPaths.size &&
-        [...nextProps.expandedPaths].every(path =>
-          nextProps.loadedProperties.has(path)
-        )) ||
-      (expandedPaths.size === nextProps.expandedPaths.size &&
-        [...nextProps.expandedPaths].some(key => !expandedPaths.has(key))) ||
-      focusedItem !== nextProps.focusedItem
+
+    const hasLoaded = [...nextProps.expandedPaths].every(path =>
+      nextProps.loadedProperties.has(path)
     );
+
+    return hasLoaded || focusedItem !== nextProps.focusedItem;
```

Comment:
so this logic is probably too naive, but it seems to work :)

1. hasLoaded will be false if you expand and things are not loaded... so it might cover all of the cases... either way, it was me goofing around and should be separate.